client/options.go

```diff
@@ -92,15 +92,24 @@ type AttachOption func(*AttachOptions)
 // AttachOptions configures how we set up the document.
 type AttachOptions struct {
 	// Presence is the presence of the client.
-	Presence innerpresence.Presence
-	IsManual bool
+	Presence    innerpresence.Presence
+	InitialRoot map[string]any
+	IsManual    bool
 }
 
 // WithPresence configures the presence of the client.
 func WithPresence(presence innerpresence.Presence) AttachOption {
 	return func(o *AttachOptions) { o.Presence = presence }
 }
 
+// WithInitialRoot sets the initial root of the document.
+// If the document already exists before attachment, this initial root will not be applied.
+func WithInitialRoot(root map[string]any) AttachOption {
```

Comment:
We could provide an option like 'ensureFirstDocument'. If this option is set to true, we can use a condition checking if ServerSeq == 1. However, I'm not certain if this is the best approach.